docs/source/tutorial/03-write-your-first-query.mdx

```diff
@@ -0,0 +1,108 @@
+<img src="images/completed_basic_query.png" class="screenshot" alt="Query JSON in GraphiQL"/>
+
+This query executes successfully, but it doesn't include any information about the `launches`! That's because we didn't include the necessary field in our query.
+
+Update your query to fetch the `id` and `site` properties for each launch, like so:
+
+```graphql:title=(GraphiQL)
+query LaunchList {
+  launches {
+    cursor
+    hasMore
+    launches {
+      id
+      site
+    }
+  }
+}
+```
+
+Run the query again, and you'll now see that in addition to the information you got back before, you're also getting a list of launches with their ID and site information:
+
+<img src="images/completed_id_query.png" class="screenshot" alt="Updated query JSON in GraphiQL"/>
+
+## Add the query to your project
+
+Now that your query is fetching the right data, head back to Android Studio.
+
+1. Right click on the `src/main/graphql/com/example/rocketserver` folder. This folder should contain your `schema.json`. Select **New > File**:
+
+<img src="images/new_file.png" class="screenshot" alt="New GraphQL file"/>
+
+2. Name the file `LaunchList.graphql`. Make sure it's saved at the same level as your `schema.json` file.
+
+3. Copy your final query from GraphiQL and paste it into `LaunchList.graphql`.
+
+```graphql:title=app/src/main/graphql/com/example/rocketreserver/LaunchList.graphql
+query LaunchList {
+  launches {
+    cursor
+    hasMore
+    launches {
+      id
+      site
+    }
+  }
+}
+```
+
+## Generate the model
+
+Build your project to have the Apollo Android plugin generate your first model. The plugin defines a task named `generateApolloSources` to generate the models. You don't need to run it. It will be executed automatically when building your project.
+
+> **Note**: autocomplete won't work until you build your project. That is because autocomplete requires the generated code to work. Each time you change your queries, you should rebuild your project for Android Studio to pick up the modifications.
+
+## Examine generated code
+
+From the menu, select **Navigate > Class** and start typing `LaunchList`, Android Studio should suggest to open `LaunchList.kt`. The file should be in `build/generated/source/apollo/debug/service/com/example/rocketreserver/LaunchListQuery.kt`.
+
+The `LaunchListQuery.kt` file defines a root class, `LaunchListQuery`, with many nested classes. If you compare the classes to the JSON data returned in GraphiQL, you see that the structure matches. These classes include properties only for the fields that your query requests.
+
+Try commenting out the `id` property in `LaunchList.graphql`, saving, then building again. When the build completes, the innermost `Launch` now only includes the built-in `__typename` and the requested `site` property.
+
+Uncomment `id` and rebuild to restore the property.
+
+Now that you've generated code and had a chance to see what's in there, it's time to [execute the query](04-execute-the-query)!
```

Comment:
Applies to all pages, I was thinking we can put the last part into its own section like `### Next step`. Not sure if it is necessary. Just an idea :)

Reply:
I took that from the iOS tutorial. I kind of like that the sentence happens in the flow and doesn't require an additional step in the right menu. Having the "next" on the right would somehow duplicate the index on the left. No real strong feeling about it though, happy to change it if that can help following the tutorial.

Reply:
Yeah, it was just an idea. If iOS is doing like this, I would say consistency is good :)